drivers/iio/adc/ad4630.c

```diff
@@ -370,26 +370,33 @@ static int ad4630_read_avail(struct iio_dev *indio_dev,
 	}
 }
 
-static int __ad4630_set_sampling_freq(const struct ad4630_state *st, unsigned int freq)
+static int __ad4630_set_sampling_freq(struct ad4630_state *st, unsigned int freq)
 {
-	struct pwm_state conv_state = {
-		.duty_cycle = 10,
-		.enabled = true,
-	}, fetch_state = {
-		.duty_cycle = 10,
-		.enabled = true,
-	};
+	struct pwm_waveform conv_wf = { }, fetch_wf = { .duty_length_ns = 10 };
```

Comment:
```suggestion
	struct pwm_waveform conv_wf = { }, fetch_wf = { };
```

This gets unconditionally written over later.

Reply:
It's actually duty_offset that gets written over. If I don't set this here, the buffer fails to initialize during testing.

Reply:
My bad.